### ISSUE TITLE
Crash when editing remote file using psedit

### ISSUE THREAD

**Issue by mbsj** [PowerShell/vscode-powershell#1091]
### System Details

- Operating system name and version: Windows 10 Pro
- VS Code version: 1.17.2
- PowerShell extension version: 1.5.0
- Output from `$PSVersionTable`:
```
Name                           Value
----                           -----
PSVersion                      5.1.14393.1770
PSEdition                      Desktop
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
BuildVersion                   10.0.14393.1770
CLRVersion                     4.0.30319.42000
WSManStackVersion              3.0
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
```
```
Copy / paste the following commands into the PowerShell Integrated Console, and paste the output here:

code -v
1.17.2
b813d12980308015bcd2b3a2f6efa5c810c33ba5

$pseditor.EditorServicesVersion
Major  Minor  Build  Revision
-----  -----  -----  --------
1      5      0      0

code --list-extensions --show-versions
codezombiech.gitignore@0.5.0
DavidAnson.vscode-markdownlint@0.11.0
donjayamanne.githistory@0.2.3
DotJoshJohnson.xml@1.9.2
eamodio.gitlens@5.7.1
humao.rest-client@0.16.0
mitchdenny.ecdc@0.10.3
ms-mssql.mssql@1.2.0
ms-vscode.csharp@1.12.1
ms-vscode.PowerShell@1.5.0
ms-vscode.Theme-MarkdownKit@0.1.4
ms-vsts.team@1.122.0
nonoroazoro.syncing@1.4.9
PKief.material-icon-theme@2.2.4
streetsidesoftware.code-spell-checker@1.4.12
Tyriar.sort-lines@1.3.0
waderyan.gitblame@2.2.0
wayou.vscode-todo-highlight@0.5.11
ziyasal.vscode-open-in-github@1.2.9

$PSVersionTable
Name                           Value
----                           -----
PSVersion                      5.1.14393.1770
PSEdition                      Desktop
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
BuildVersion                   10.0.14393.1770
CLRVersion                     4.0.30319.42000
WSManStackVersion              3.0
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
```

### Issue Description

Every time i try to edit a file in a remote session the integrated console crashes. 
I simple create and enter a new session (Enter-PSSession) then i use psedit to edit the file. After a few seconds, maybe as much as a minute or two, the process crashes. 
I have disabled all extensions and the issue persists. 

### Attached Logs
[1509974465-82c81cab-97f9-4af7-a119-16c6ff0a97661509974457601.zip](https://github.com/PowerShell/vscode-powershell/files/1446320/1509974465-82c81cab-97f9-4af7-a119-16c6ff0a97661509974457601.zip)

**Comment by rkeithhill** [comment]
What version of PowerShell is on the other end?  This seems to be crashing with a PSNotSupportedException when PSES is asking for all the commands of type alias on the remote side.  Is it a constrained endpoint?

**Comment by rkeithhill** [comment]
Well, this would `splain the exception:

https://github.com/PowerShell/PowerShell/blob/4bc52d2358222084738157a08907fac32d13bd3a/src/System.Management.Automation/engine/remoting/client/remoterunspace.cs#L2982-L2988

Was the remote session a PowerShell Core session?

**Comment by mbsj** [comment]
Unfortunately it's not PowerShell core. I'm connecting to a Server 2008 R2 with version 4. Not constrained.

**Comment by mbsj** [comment]
I just made another test, remoting to a server running PowerShell 5.1. It behaves the same way with the same error.

**Comment by rkeithhill** [comment]
If you look at the EditorServices.log file for that session, do you see the same PSNotSupportedException?  I can only see the source code for PS Core (since it's OSS).  But peeking into the SMA.dll v3 with dotPeek, I see that the same method throws PSNotSupportedException as well.  Hmm, I wonder if we're using the right type for remote sessions...

**Comment by mbsj** [comment]
Yes, the exact same error with GetAliases. I'm away from my PC right now but I can attach logs for a session to the 5.1 instance later, if you'd like?

**Comment by mbsj** [comment]
For the sake of being thorough, i have a attached the logs for the 5.1 session as well. 
[1510215916-ede6715c-2016-46b6-a6d4-0085661f6d1e1510215883874.zip](https://github.com/PowerShell/vscode-powershell/files/1456883/1510215916-ede6715c-2016-46b6-a6d4-0085661f6d1e1510215883874.zip)

**Comment by rkeithhill** [comment]
Thanks.  Yup, same exception.  Have a fix for that particular exception.

**Comment by rkeithhill** [comment]
This should fixed in the upcoming 1.5.1 release.  If it isn't, please re-open this issue.
